Description: Override upstream copy and use libbsd
  Avoid using local copy of strlcpy, use the libbsd library in overlay mode
  instead on our supported targets.
  .
  Updated by Otto Kekäläinen after 3.3 import to match the situation at 2.6.
Author: Yuri D'Elia <wavexx@thregr.org>
Forwarded: not-needed
Updated: 2015-11-29

diff --git a/Makefile.debian b/Makefile.debian
new file mode 100644
index 0000000..1697026
--- /dev/null
+++ b/Makefile.debian
@@ -0,0 +1,12 @@
+# `entr` provides a built-in compability layer for Linux. To use compatibility
+# libraries instead insteall libkqueue and libbsd
+#
+# sudo apt-get install libkqueue-dev libbsd-dev pkg-config
+# make test -f Makefile.linux-lbsd
+# sudo make install
+
+CFLAGS += -D_GNU_SOURCE -I/usr/include/kqueue $(shell pkg-config --cflags libbsd-overlay)
+LDFLAGS += -lpthread -lkqueue $(shell pkg-config --libs libbsd-ctor)
+MANPREFIX ?= ${PREFIX}/share/man
+
+include Makefile.bsd
diff --git a/Makefile.kfreebsd b/Makefile.kfreebsd
new file mode 100644
index 0000000..e0f02de
--- /dev/null
+++ b/Makefile.kfreebsd
@@ -0,0 +1,5 @@
+CFLAGS += -D_GNU_SOURCE $(shell pkg-config --cflags libbsd-overlay)
+LDFLAGS += $(shell pkg-config --libs libbsd-overlay)
+MANPREFIX ?= ${PREFIX}/share/man
+
+include Makefile.bsd
diff --git a/Makefile.linux b/Makefile.linux
index c8f7c3b..8bcfde8 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -1,5 +1,6 @@
-CPPFLAGS += -D_GNU_SOURCE -D_LINUX_PORT -Imissing
+CPPFLAGS += -D_GNU_SOURCE -Imissing $(shell pkg-config --cflags libbsd-overlay)
+LDFLAGS += $(shell pkg-config --libs libbsd-ctor)
 MANPREFIX ?= ${PREFIX}/share/man
-EXTRA_SRC = missing/strlcpy.c missing/kqueue_inotify.c
+EXTRA_SRC = missing/kqueue_inotify.c
 
 include Makefile.bsd

